roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -229,54 +492,46 @@ def __init__(self, filepath, optical_channel_name=None,
         # self.image_masks = np.moveaxis(np.array(ps['image_mask'].data), [0, 1, 2], [2, 0, 1])
         if 'image_mask' in ps.colnames:
             self.image_masks = DatasetView(ps['image_mask'].data).lazy_transpose([1, 2, 0])
-        if 'pixel_mask' in ps.colnames:
-            # Extract pixel_mask/background:
-            px_list = [ps['pixel_mask'][e] for e in range(ps['pixel_mask'].data.shape[0])]
-            temp = np.empty((1, 4))
-            for v, b in enumerate(px_list):
-                temp = np.append(temp, np.append(b, v * np.ones([b.shape[0], 1]), axis=1), axis=0)
-            self.pixel_masks = temp[1::, :]
         if 'RoiCentroid' in ps.colnames:
             self._roi_locs = ps['RoiCentroid']
         if 'Accepted' in ps.colnames:
             self._accepted_list = ps['Accepted'].data[:]
         # Extract Image dimensions:
 
         # Extract roi_response:
-        self.roi_resp_dict = dict()
-        self._roi_names = [_nwbchildren_name[val]
+        roi_names = [_nwbchildren_name[val]
                       for val, i in enumerate(_nwbchildren_type) if i == 'RoiResponseSeries']
-        if not self._roi_names:
+        if not roi_names:
             raise Exception('no ROI response series found')
         else:
-            for roi_name in self._roi_names:
-                self.roi_resp_dict[roi_name] = mod['Fluorescence'].get_roi_response_series(roi_name)
-        self.roi_response = self.roi_resp_dict[self._roi_names[0]]
+            for trace_name in ['roiresponseseries', 'dff', 'neuropil', 'deconvolved']:
+                trace_name_nwb = [j for j,i in enumerate(roi_names) if trace_name in i.lower()]
+                if trace_name_nwb:
+                    trace_name_segext = 'raw' if trace_name == 'roiresponseseries' else trace_name
+                    setattr(self,f'_roi_response_{trace_name_segext}',
+                            mod['Fluorescence'].get_roi_response_series(roi_names[trace_name_nwb[0]]).data[:].T)
 
         # Extract samp_freq:
-        self._samp_freq = self.roi_response.rate
-        # Extract no_rois/ids:
+        self._sampling_frequency = mod['Fluorescence'].get_roi_response_series(roi_names[0]).rate
+        # Extract get_num_rois()/ids:
         self._roi_idx = np.array(ps.id.data)
 
         # Imaging plane:
         _optical_channel_exist = [i for i, e in enumerate(
```

Comment:
Optical channels will be in `ophys/OpticalChannel`.

Why not just looping through those:
```
if `OpticalChannel` in nwbfile['ophys']:
    for oc in nwbfile['ophys']['OpticalChannel]:
        ...
```
 I think it's nicer to read!